bpf/lib/tailcall.h

```diff
@@ -103,14 +102,28 @@
 /* Private per-EP map for internal tail calls. Its bpffs pin is replaced every
  * time the BPF object is loaded. An existing pinned map is never reused.
  */
-struct bpf_elf_map __section_maps cilium_calls = {
-	.type		= BPF_MAP_TYPE_PROG_ARRAY,
-	.id		= CILIUM_MAP_CALLS,
-	.size_key	= sizeof(__u32),
-	.size_value	= sizeof(__u32),
-	.pinning	= CILIUM_PIN_REPLACE,
-	.max_elem	= CILIUM_CALL_SIZE,
-};
+struct {
+	__uint(type, BPF_MAP_TYPE_PROG_ARRAY);
+	__uint(key_size, sizeof(__u32));
+	__uint(max_entries, CILIUM_CALL_SIZE);
+	__uint(pinning, CILIUM_PIN_REPLACE);
+	__array(values, int ());
+} cilium_calls __section(".maps");
```

Comment:
Nit: I guess we want to use `__section_maps_btf` just like we do everywhere else.

Reply:
:facepalm:

Reply:
Also realized __section_maps can be removed now.